zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Schema.scala

```diff
@@ -39,42 +39,93 @@ object Relation extends Enumeration {
   val ~= = Value("~=")
 }
 
-class Field(val name: String, val dataType: DataType, val isOptional: Boolean = false) {
+trait Field {
+  val name: String
+  val dataType: DataType
+  val isOptional: Boolean = false
+
+  def as(name: String): Field
```

Comment:
Hmm...This is just used for cloning into a new field with different name, but with all other members the same. Basically the same semantics with `as` keyword.